pkg/sql/opt/norm/testdata/rules/scalar

```diff
@@ -222,7 +222,7 @@ project
       ├── a.s:4::VARCHAR(2) [as=s:11, outer=(4), immutable]
       ├── a.i:2::INT2::INT8 [as=i:12, outer=(2), immutable]
       ├── a.s:4::CHAR::VARCHAR [as=s:13, outer=(4), immutable]
-      ├── ARRAY[a.i:2, 2]::OIDVECTOR [as=array:14, outer=(2), stable]
```

Comment:
This is changing because the cast from INT to OID is now marked as immutable. As far as I can tell, it _is_ immutable in Postgres, but I don't know why it wasn't set as such before. Notice the `ignoreVolatilityCheck` is set [here](https://github.com/RaduBerinde/cockroach/blob/123ae8783d0c63e935380bed946a44cd8d332334/pkg/sql/sem/tree/casts.go#L255) because we were diverging from Postgres's volatility of this cast.

Reply:
Ahh... REG* types are part of the OID family ([which they really shouldn't be](https://github.com/cockroachdb/cockroach/blob/ce2796a987266f56fd1adcc1d9cc9aefe7c92c39/pkg/sql/types/types.proto#L185-L186)) and INT -> REG* types are stable. INT -> OID was previously marked as stable because volatilities were labeled at the family level. Now that volatilities can be labeled at the type OID level, INT->OID can be correctly marked as immutable.